src/subclass/simple.rs

```diff
@@ -56,7 +56,7 @@ unsafe impl<T: ObjectSubclass> super::types::ClassStruct for ClassStruct<T> {
 }
 
 impl<T: ObjectSubclass> ops::Deref for ClassStruct<T> {
```

Comment:
Is this still needed? That deref would just go via `Class<T>` now or not?

Reply:
Hm. Well `ClassStruct` is different from `Class`. I don't see anywhere else `Deref` is implemented for `ClassStruct`, so unless I'm missing something it won't deref without this.

Reply:
But where is `simple::ClassStruct` used if not as part of `klass_init()` where you would now pass a `Class<simple::ClassStruct<_>>` or not?

Reply:
Basically `simple::ClassStruct` is an FFI class struct currently, so it would seem more consistent to use the `Class` wrapper around it in all code that uses it as such

Reply:
I'm not entirely familar with how the `ClassStruct` and `InstanceStruct` are used.

But `ObjectSubclass::class_init()` seems to take `Self::Class` as an argument, which is something implementing the `ClassStruct` trait, usually `simple::ClassStruct`. Not the new `Class` type.

I don't believe `simple::ClassStruct` implements `ObjectType`, so `Class<simple::ClassStruct<_>>` isn't a valid type anyway.

Reply:
Indeed. Let's keep that for later then to clean this up, but can you create an issue for this?